enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,184 @@
+---
+title: container-runtime-containerd
+authors:
+  - "@selansen"
+reviewers:
+  - "@aravindhp"
+approvers:
+  - "@aravindhp"
+  - 
+creation-date: 2021-11-19
+last-updated: 2021-11-20
+status: implementable
+---
+
+# containerd - new container run time  
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow customers to bring up windows node with containerd
+as default run time from kubernetes 1.24 based openshift 4.11 onwards. When customers try to upgrade 
+current cluster to kubernetes 1.24 based openshift 4.11, runtime will be migrated from docker to containerd
+
+## Motivation
+
+In Kubernetes, the CRI interface is used to talk to a container runtime, The design of CRI is to be able
+to run a CRI implementation as a separate binary. However currently the CRI of docker (a.k.a. dockershim) 
+is part of kubelet code, runs as part of kubelet and is tightly coupled with kubelet's lifecycle. From kubernetes
+1.24 onwards dockershim will be removed from kubelet code. Currently WMCO uses docker as default run time. Aim is
+to make containerd as default runtime and move away from docker before dokershim has been decoupled from kubelet.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Make containerd as default run time. 
+* When upgrade happens from older release newer one, containerd will become default run time.  
+
+### Non-Goals
+
+* De-configuring docker run time is not part of this enhancement.
+* Refactoring or re-design of WMCO due to dockershim deprecation.
+* Switching between docker and containerd runtimes is not supported.
+
+## Proposal
+
+To make containerd as default run time, containerd should be installed before kueblet and 
+kubelet parameter need to be updated so that containerd will become default run time. In 
+upgrade case each Windows VM configured by previous versions of the WMCO has its Machine 
+object deleted, resulting in the drain and deletion of the Windows node, and the termination
+then followed by recreation of the VM. The upgraded WMCO instance will then be able to configure 
+VMs that will be created with containerd as default runtime. All minimum requirement stated 
+as part of  WMCO upgrade will be applicable here as well.
+
+Containerd Migration plan
+    * Containerd will become default runtime as part of openshift 4.11 
+    * For current usage, we plan to introduce feature flag and release it only for community operator.
```

Comment:
Please make it clear if users will have access to this flag or if it will only be used in development.
Also its not clear the default state of the flag for the community release